testsuite/pytests/test_spatial/test_SynapseCollection_distance.py

```diff
@@ -23,159 +23,152 @@
 Tests distance between sources and targets of SynapseCollection
 """
 
-import unittest
+import pytest
 import math
 import nest
 
 
-class SynapseCollectionDistance(unittest.TestCase):
+@pytest.fixture(autouse=True)
+def _reset_kernel():
+    nest.ResetKernel()
 
-    def setUp(self):
-        nest.ResetKernel()
 
-    def calculate_distance(self, conns, s_nodes, t_nodes):
-        """Calculate a reference distance between source and target nodes"""
+def _calculate_distance(conns, s_nodes, t_nodes):
+    """Calculate a reference distance between source and target nodes"""
 
-        s_pos = nest.GetPosition(s_nodes)
-        t_pos = nest.GetPosition(t_nodes)
+    s_pos = nest.GetPosition(s_nodes)
+    t_pos = nest.GetPosition(t_nodes)
 
-        src = conns.source
-        trgt = conns.target
+    src = conns.source
+    trgt = conns.target
 
-        in_3d = len(s_pos[0]) == 3
+    in_3d = len(s_pos[0]) == 3
 
-        ref_distance = []
-        for s, t in zip(src, trgt):
-            x_ref = t_pos[t_nodes.index(t)][0] - s_pos[s_nodes.index(s)][0]
-            y_ref = t_pos[t_nodes.index(t)][1] - s_pos[s_nodes.index(s)][1]
-            z_ref = 0.0
-            if in_3d:
-                z_ref = t_pos[t_nodes.index(t)][2] - s_pos[s_nodes.index(s)][2]
+    ref_distance = []
+    for s, t in zip(src, trgt):
+        x_ref = t_pos[t_nodes.index(t)][0] - s_pos[s_nodes.index(s)][0]
+        y_ref = t_pos[t_nodes.index(t)][1] - s_pos[s_nodes.index(s)][1]
+        z_ref = 0.0
+        if in_3d:
+            z_ref = t_pos[t_nodes.index(t)][2] - s_pos[s_nodes.index(s)][2]
 
-            ref_dist = math.sqrt(x_ref * x_ref + y_ref * y_ref + z_ref * z_ref)
-            ref_distance.append(ref_dist)
+        ref_dist = math.sqrt(x_ref * x_ref + y_ref * y_ref + z_ref * z_ref)
+        ref_distance.append(ref_dist)
```

Comment:
Thanks for the suggestion. I took it yet a bit further, see my new push. All other changes I would leave for a later PR so that we can get this bug out of the way.